src/geometry/ArborX_GeometryTraits.hpp

```diff
@@ -52,12 +36,39 @@ struct tag
 {
   using type = not_specialized;
 };
+template <typename Geometry>
+using tag_t = typename tag<Geometry>::type;
 
 template <typename Geometry>
 struct coordinate_type
 {
   using type = not_specialized;
 };
+template <typename Geometry>
+using coordinate_type_t = typename coordinate_type<Geometry>::type;
+
+// clang-format off
+#define DEFINE_GEOMETRY(name, name_tag)                                        \
+  struct name_tag{};                                                           \
+  template <typename Geometry>                                                 \
+  struct is_##name : std::is_same<tag_t<Geometry>, name_tag>{};              \
```

Comment:
What's the point of defining the trait if you don't use it.  You could define directly the helper variable template.

Reply:
Technically, `ArborX::GeometryTraits` is a user visible namespace. Removing `is_point` and such is not backwards compatible.

Reply:
It is still somewhat experimental and it is not documented.  Also that is not a type that is intended to be specialized for user-defined geometry types.
I am not saying that we absolutely have to drop the trait type but that I think that there is nothing that prevent us to do so.